mlir/lib/Dialect/Rock/IR/RockDialect.cpp

```diff
@@ -499,6 +499,14 @@ template <typename T> static LogicalResult verifyConvOp(T op) {
       isDisjointed("input_layout", "hi", "wi"))
     return op.emitError("Disjointed yx or hw!");
 
+  bool isXdlops = bitEnumContainsAll(
+      op->getAttr("features").template cast<GemmFeaturesAttr>().getValue(),
+      GemmFeatures::mfma);
+  if (op->hasAttr("derivedBlockSize") && !isXdlops) {
```

Comment:
Instead of `hasAttr` by string, maybe use a higher-level call like `derivedBlockSizeAttr()` and checking for `none`

Reply:
I'll also note that I'd prefer the different condition where the error is to have `derivedBlockSize` and a tuning parameter attribute that has a `GeneralGridwiseGemmParams`

Reply:
It's Operation* here, can't cast it either.

Reply:
I'd argue for rewriting this to a combination of the conv interface and the gemm wrapper interface (especially since we can safely cast from the conv interface to the gemm interface without worrying much because we know the deal)

Reply:
I tried it first using 
RockConvInterface convInterface = cast<RockConvInterface>(op);
but compiler still treated convInterface as Operation* and threw error.

Anything  else I can try?